Fails to compile on nightly 2018-04-27 with nightly feature enabled

Trying to compile with current nightly toolchain (2018-04-27) with nightly feature enabled fails with:  
```
error[E0425]: cannot find function `unreachable` in module `std::mem`
  --> /home/foo/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.0.0/src/nightly_lazy.rs:30:35
   |
30 |                 None => std::mem::unreachable(),
   |                                   ^^^^^^^^^^^ not found in `std::mem`
help: possible candidate is found in another module, you can import it into scope
   |
10 | use lazy::std::intrinsics::unreachable;
   |

```

Looks like a regression on nightly as it compiles fine with nightly-2018-04-10.

edit: fix nightly version date

--- a/src/nightly_lazy.rs
+++ b/src/nightly_lazy.rs
@@ -6,6 +6,7 @@
 // copied, modified, or distributed except according to those terms.
 
 extern crate std;
+extern crate core;
 
 use self::std::prelude::v1::*;
 use self::std::sync::Once;
@@ -27,7 +28,7 @@ impl<T: Sync> Lazy<T> {
         unsafe {
             match self.0 {
                 Some(ref x) => x,
-                None => std::mem::unreachable(),
+                None => core::hint::unreachable_unchecked(),
             }
         }
     }

